CISLibrary/drivers/HBridge/DRV8837.cpp

```diff
@@ -7,51 +7,38 @@
  * @param rev pin
  * @param sleep pin
  */
-DRV8837::DRV8837(PinName fwd, PinName rev, PinName nSleep)
-  : fwd(fwd), rev(rev), nSleep(nSleep)
-{
-  // set the pwm frequencies
-  this->fwd.period(1.0 / DRV8837_PWM_FREQ_HZ);
-  this->rev.period(1.0 / DRV8837_PWM_FREQ_HZ);
-
-  // disable initially
-  this->nSleep.write(1);
+DRV8837::DRV8837(PinName fwd, PinName rev, PinName nSleep) :
+  fwd(fwd), rev(rev), nSleep(nSleep, 0) {
+  // set the pwm periods
+  this->fwd.period_us(DRV8837_PWM_PERIOD_US);
+  this->rev.period_us(DRV8837_PWM_PERIOD_US);
 }
 
 /**
  * @brief Destroy the DRV8837::DRV8837 object
  *
  */
-DRV8837::~DRV8837() { }
+DRV8837::~DRV8837() {}
 
 /**
  * @brief Set the output of the HBridge
  *
  * @param value [-1.0, 1.0]
- * @param blocking will wait until operation is complete if true
+ * @param blocking parameter ignored on the DRV8837 implementation
  * @return mbed_error_status_t
  */
-mbed_error_status_t DRV8837::set(double value) {
-  if (value < -1.0 || value > 1.0) {
-    return MBED_ERROR_CODE_INVALID_ARGUMENT;
-  }
+mbed_error_status_t DRV8837::set(double value, bool blocking) {
+  this->setSleep(false);
```

Comment:
Don't touch sleep in here. The function just sets the output not the sleep status.
For the unused bool blocking, it could give you an used warning, you could comment it out: bool /* blocking */